### PR TITLE
Allow creating outputs file when simulation starts at non-zero time

### PR DESCRIPTION
**Pull request by JeenaYun**
The changes are made to allow the creation of probe outputs when the simulation starts at non-zero time. This feature might be useful in the case of loading checkpoints.

### CODE REVIEW

File: src/io/BoundaryProbeWriter.cpp [hunk 1]
```diff
@@ -83,7 +85,17 @@ void BoundaryProbeWriter<D>::write(double time,
             out_->open(probe.file_name, false);
             write_header(probe, functions);
         } else {
-            out_->open(probe.file_name, true);
+            fs::path pckp(probe.file_name);
+            bool exists = fs::exists(pckp);
+            if (exists) {
+                // open existing file
```
Comment (hpc4geo) [inline src/io/BoundaryProbeWriter.cpp:91]: Please remove all these comments

File: src/io/ScalarWriter.cpp [hunk 2]
```diff
@@ -19,7 +21,17 @@ void ScalarWriter::write(double time, mneme::span<double> scalars) const {
         out_->open(file_name_, false);
         write_header();
     } else {
-        out_->open(file_name_, true);
+        fs::path pckp(file_name_);
+        bool exists = fs::exists(pckp);
+        if (exists) {
+            // open existing file
```
Comment (hpc4geo) [inline src/io/ScalarWriter.cpp:27]: Please delete all these comments

File: src/io/BoundaryProbeWriter.cpp [hunk 3]
```diff
@@ -83,7 +85,17 @@ void BoundaryProbeWriter<D>::write(double time,
             out_->open(probe.file_name, false);
             write_header(probe, functions);
         } else {
-            out_->open(probe.file_name, true);
+            fs::path pckp(probe.file_name);
```
Comment (hpc4geo) [inline src/io/BoundaryProbeWriter.cpp:88]: Remove lines 88-90 and replace with just

```
if (std::filesystem::exists(probe.file_name)) {
```

File: src/io/BoundaryProbeWriter.cpp [hunk 4]
```diff
@@ -8,6 +8,8 @@
 #include <sstream>
 #include <unordered_map>
 #include <unordered_set>
+#include <filesystem>
+namespace fs = std::filesystem;
```
Comment (hpc4geo) [inline src/io/BoundaryProbeWriter.cpp:12]: Remove line 12 (uneeded - see below)

File: src/io/ScalarWriter.cpp [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 #include <iomanip>
 #include <ios>
+#include <filesystem>
+namespace fs = std::filesystem;
```
Comment (hpc4geo) [inline src/io/ScalarWriter.cpp:6]: Remove line 6

File: src/io/ScalarWriter.cpp [hunk 6]
```diff
@@ -19,7 +21,17 @@ void ScalarWriter::write(double time, mneme::span<double> scalars) const {
         out_->open(file_name_, false);
         write_header();
     } else {
-        out_->open(file_name_, true);
+        fs::path pckp(file_name_);
+        bool exists = fs::exists(pckp);
+        if (exists) {
```
Comment (hpc4geo) [inline src/io/ScalarWriter.cpp:26]: Replace lines 24-26 with 
```
if (std::filesystem::exists(file_name_)) {
```

**Review by hpc4geo — CHANGES_REQUESTED** [review]
Please update the code

**Comment by JeenaYun** [comment]
Hi, thanks for the comment. I reflected your suggestions.

**Comment by dmay23** [comment]
You have the wrong file name in ScalarWritwr.cpp. You refer to
probe.file_name in the if() statement.

Always check the code compiles before committing and pushing.

**Comment by JeenaYun** [comment]
I followed the convention used inside the file ScalarWriter.cpp. The variable name pointing to the file name is already `file_name_` not `probe.file_name` even before my commit. I guess it is because ScalarWriter.cpp is not related to 'probe' outputs, but rather used for fault scalar output. Correct me if I am wrong.

**Comment by hpc4geo** [comment]
> I followed the convention used inside the file ScalarWriter.cpp. The variable name pointing to the file name is already `file_name_` not `probe.file_name` even before my commit. I guess it is because ScalarWriter.cpp is not related to 'probe' outputs, but rather used for fault scalar output. Correct me if I am wrong.

```
symgrad:~/codes/tandem/src/io % grep probe ScalarWriter.cpp  
symgrad:~/codes/tandem/src/io %                                             <dmay/seas-checkpoint>
```
There is no reference to anything called `probe` in ScalarWriter.cpp

**Comment by JeenaYun** [comment]
Oh I see what you mean. That was a typo, sorry.

**Comment by dmay23** [comment]
Attempting to compile the code would have exposed this typo….

**Review by hpc4geo — APPROVED** [review]
Thanks for your contribution Jeena.

**Pull request merged**